pages/writing.md

```diff
@@ -45,23 +45,131 @@ juliaup status
 
 ## Development environments
 
+Depending on your experience and how you use Julia, some ways of writing code may be more familiar or useful to you than others.
+Data scientists commonly prefer notebook environments such as those provided by [Jupyter](https://jupyter.org/) through the [IJulia.jl](https://github.com/JuliaLang/IJulia.jl) package, or Julia's own reactive notebooks from [Pluto.jl](https://plutojl.org/).
+On the other hand, package developers or those who write a lot of scripts will prefer the more traditional programming environment provided by a text editor or integrated development environment (IDE).
+
+For some, the Julia REPL (Read Evaluate Print Loop) itself is all that's necessary.
+The REPL's primary function is to run code in "Julian" mode, but it also has a number of other modes that expand what can be done from inside Julia.
+Each mode is enterred by typing a character into the REPL from Julian mode, and can be exited by deleting this character with backspace.
+
+### Help mode (`?`)
+By entering a `?`, you can query information and metadata about Julia objects and unicode symbols simply by typing their name into the command line.
+For functions, types, and variables, the query fetches things such as documentation, type fields and supertypes, and in which file the object is defined.
+<!-- How do you do syntax highlighting for the Julia REPL? -->
+```
+help?> Int
+search: Int Int8 Int64 Int32 Int16 Int128 Integer intersect intersect!
+
+  Int64 <: Signed
+
+
+  64-bit signed integer type.
+```
+
+For unicode symbols, the query will return how to type the symbol in the REPL, which is useful when you copy-paste a symbol in without knowing its name, and fetch information about the object the symbol is bound to, just as above.
+
+### Pkg mode (`]`)
+Pkg mode is for managing environments and packages that is based on the Rust package manager "cargo".
```

Comment:
I think this sentence is missing a word.

Reply:
(I'm also not sure the comparison to Rust's cargo is crucial enough to belong in the first sentence. This might confuse people who have no experience with Rust.)